momento-sdk/src/main/java/momento/sdk/messages/CacheSortedSetFetchResponse.java

```diff
@@ -0,0 +1,98 @@
+package momento.sdk.messages;
+
+import grpc.cache_client._SortedSetElement;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.NavigableSet;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+import momento.sdk.exceptions.SdkException;
+
+/** Response for a sorted set fetch operation */
+public interface CacheSortedSetFetchResponse {
+
+  /** A successful sorted set fetch operation that found elements. */
+  class Hit implements CacheSortedSetFetchResponse {
+    private final List<_SortedSetElement> elements;
+
+    /**
+     * Constructs a sorted set fetch hit with a list of encoded elements.
+     *
+     * @param elements the retrieved elements.
+     */
+    public Hit(List<_SortedSetElement> elements) {
+      this.elements = elements;
+    }
+
+    public NavigableSet<ScoredElement<String>> elementsSet() {
```

Comment:
I added these navigable set and ordered list methods to compare with the ordered map method.

```java
{
  // The map won't update its order if the scores are changed
  final Map<String, Double> scoredMap = hit.elementsMap();
  // iterate through in order
  for (Map.Entry<String, Double> entry : scoredMap.entrySet()) {
    final String element = entry.getKey();
    final double score = entry.getValue();
    // ...
  }

  // get score (doesn't work for byte[])
  final Double score = scoredMap.get("thing I want");
}

{
  // The list won't update its order if the scores are changed
  final List<ScoredElement<String>> scoredList = hit.elementsList();
  // iterate through in order
  for (ScoredElement<String> scoredElement : scoredList) {
    final String element = scoredElement.getElement();
    final double score = scoredElement.getScore();
    // ...
  }

  // get score
  final Optional<Double> score =
      scoredList.stream()
          .filter(se -> se.getElement().equals("thing I want"))
          .map(ScoredElement::getScore)
          .findFirst();
}

{
  // Will maintain order when modifying set. Doesn't make sense to specify
  // ascending/descending in the method call because this can be traversed either way.
  final NavigableSet<ScoredElement<String>> scoredSet = hit.elementsSet();
  // iterate through in ascending order
  for (ScoredElement<String> scoredElement : scoredSet) {
    final String element = scoredElement.getElement();
    final double score = scoredElement.getScore();
    // ...
  }

  // iterate through in descending order
  for (ScoredElement<String> scoredElement : scoredSet.descendingSet()) {
    final String element = scoredElement.getElement();
    final double score = scoredElement.getScore();
    // ...
  }

  // get score
  final Optional<Double> score =
      scoredSet.stream()
          .filter(se -> se.getElement().equals("thing I want"))
          .map(ScoredElement::getScore)
          .findFirst();
}
```

Reply:
@cprice404 would you like a more in-depth example?

Reply:
no this is great, thanks.

One thing to call out here is that I don't know if these `stream` parts are that relevant; if the user wants a score for an individual element they would probably just call one of the other methods like `GetScore`/`GetScores`.  If they want to do something will all the scores then they would just access the score of the current element inside of a loop or `map`.  So personally I wouldn't weigh that stanza very heavily in evaluating the ergonomics of each of these examples.

Given that, I really like your `NavigableSet` solution.  That has my vote.